src/components/DefaultContainer/DefaultContainer.tsx

```diff
@@ -70,40 +84,117 @@ export const DefaultContainer = () => {
     setIsVisibleAddMailModal((prevState: boolean) => !prevState);
   };
 
-  const navigateToDashboard = (props: RouteComponentProps) => (
-    <Dashboard
-      {...props}
-      onGetUserProfile={handleGetUserProfile}
-      spidMail={userProfile.email}
-      onWorkMailSet={handleWorkMailSet}
-      toggleAddMailModal={toggleAddMailModal}
+  /**
+   * Given a cookie key `cookieName`, returns the value of
+   * the cookie or empty string, if the key is not found.
+   */
+  function getCookie(cookieName: string): string {
+    return (
+      document.cookie
+        .split(";")
+        .map(c => c.trim())
+        .filter(cookie => {
+          return (
+            cookie.substring(0, cookieName.length + 1) === `${cookieName}=`
+          );
+        })
+        .map(cookie => {
+          return decodeURIComponent(cookie.substring(cookieName.length + 1));
+        })[0] || ""
+    );
+  }
+
+  /**
+   * Set token in token context
+   */
+  useEffect(() => {
+    const token = getCookie("sessionToken");
+    // if getCookie returns an empty string and the user is not in pre login page (where token still has to be set),
+    // it means the token has expired and user browser deleted it -> redirect user to login, otherwise set tokenContext.token
+    if (
+      NonEmptyString.is(token) ||
+      location.pathname === "/spid-login" ||
+      customWindow._env_.IO_ONBOARDING_PA_IS_MOCK_ENV === "1"
+    ) {
+      tokenContext.setToken(token);
+    }
+    // TODO: when available, redirect to logout page (tracked in story https://www.pivotaltracker.com/story/show/169033467)
+    else {
+      props.history.push("/home");
+    }
+  }, [location.pathname]);
+
+  useEffect(() => {
+    // make api call only after onMount because token is string in any case, no longer undefined, and only if userProfile is not set and user is not on spid login page
+    if (
+      NonEmptyString.is(tokenContext.token) &&
+      !NonEmptyString.is(userProfile.given_name) &&
+      location.pathname !== "/spid-login"
+    ) {
+      const url =
+        customWindow._env_.IO_ONBOARDING_PA_API_HOST +
+        ":" +
+        customWindow._env_.IO_ONBOARDING_PA_API_PORT +
+        "/profile";
+      fetch(url, {
```

Comment:
please add a todo comment linked to pivotal story to refactor these calls with io.utils generated code